### ISSUE TITLE
Add test adapter definitions for blacklist tests.

### ISSUE THREAD

**Issue by scheib** [WebBluetoothCG/web-bluetooth#214]
Changes such as https://codereview.chromium.org/1666003002/ and subsequent ones are adding additional Adapter definitions to support testing blacklist features. Once stable, add the necessary test concepts to https://webbluetoothcg.github.io/web-bluetooth/tests/

**Comment by scheib** [comment]
Initial requestDevice test needs:

```
  // |HeartRateAndHIDAdapter|
  // Inherits from |EmptyAdapter|
  // Internal Structure:
  //   - |ConnectableDevice|(adapter, "Heart Rate And HID Device", uuids)
  //      - Advertised UUIDs:
  //         - Heart Rate UUID (0x180d)
  //         - Human Interface Device UUID (0x1812) (a blacklisted service)
  //      - Services:
  //         - Generic Access Service - Characteristics as described in
  //           GetGenericAccessService.
  //         - Heart Rate Service - Characteristics as described in
  //           GetHeartRateService.
  //         - Human Interface Device Service - No characteristics needed
  //           because the service is blacklisted.

```

**Comment by scheib** [comment]
Updating with the addition of the [getCharacteristic test](https://codereview.chromium.org/1697873002/).
```
  // |HeartRateAndHIDAdapter|
  // Inherits from |EmptyAdapter|
  // Internal Structure:
  //   - |ConnectableDevice|(adapter, "Heart Rate And HID Device", uuids)
  //      - Advertised UUIDs:
  //         - Heart Rate UUID (0x180d)
  //         - Human Interface Device UUID (0x1812) (a blacklisted service)
  //      - Services:
  //         - Device Information Service - Characteristics as described in
  //           GetDeviceInformationService.
  //         - Generic Access Service - Characteristics as described in
  //           GetGenericAccessService.
  //         - Heart Rate Service - Characteristics as described in
  //           GetHeartRateService.
  //         - Human Interface Device Service - No characteristics needed
  //           because the service is blacklisted.

  // |GetDeviceInformationService|
  // Internal Structure:
  //  - Characteristics:
  //     - Serial Number String: (a blacklisted characteristic)
  //        - Mock Functions:
  //           - Read: Fails test.
  //           - GetProperties: Returns
  //               BluetoothGattCharacteristic::PROPERTY_READ

```

**Comment by scheib** [comment]
Updating with the addition of [readValue and writeValue](https://codereview.chromium.org/1737423002/)

(renaming HeartRateAndHIDAdapter to BlacklistTestAdapter)
```
  // |BlacklistTestAdapter|
  // Inherits from |EmptyAdapter|
  // Internal Structure:
  //   - |ConnectableDevice|(adapter, "Blacklist Test Device", uuids)
  //      - UUIDs:
  //         - Blacklist Test Service UUID
  //           (611c954a-263b-4f4a-aab6-01ddb953f985)
  //         - Device Information UUID (0x180a)
  //         - Generic Access UUID (0x1800)
  //         - Heart Rate UUID (0x180d)
  //         - Human Interface Device UUID (0x1812) (a blacklisted service)
  //      - Services:
  //         - Blacklist Test Service - Characteristics as described in
  //           GetBlacklistTestService.
  //         - Device Information Service - Characteristics as described in
  //           GetDeviceInformationService.
  //         - Generic Access Service - Characteristics as described in
  //           GetGenericAccessService.
  //         - Heart Rate Service - Characteristics as described in
  //           GetHeartRateService.
  //         - Human Interface Device Service - No characteristics needed
  //           because the service is blacklisted.

  // |BlacklistTestService|
  // Internal Structure:
  //  - Characteristics:
  //     - Blacklist Exclude Reads Characteristic:
  //       (bad1c9a2-9a5b-4015-8b60-1579bbbf2135)
  //        - Mock Functions:
  //           - Read: Fails test.
  //           - Write: Calls success callback.
  //           - GetProperties: Returns
  //               BluetoothGattCharacteristic::PROPERTY_READ |
  //               BluetoothGattCharacteristic::PROPERTY_WRITE

  // |DeviceInformationService|
  // Internal Structure:
  //  - Characteristics:
  //     - Serial Number String: (a blacklisted characteristic)
  //        - Mock Functions:
  //           - Read: Fails test.
  //           - GetProperties: Returns
  //               BluetoothGattCharacteristic::PROPERTY_READ

  // |GenericAccessService|
  // Internal Structure:
  //  - Characteristics:
  //  add: 
  //     - Peripheral Privacy Flag: (0x2A02) (blacklisted for writes)
  //        - Mock Functions:
  //           - Read: Calls success callback with boolean value 'false'.
  //           - Write: Fails test.
  //           - GetProperties: Returns
  //               BluetoothGattCharacteristic::PROPERTY_READ |
  //               BluetoothGattCharacteristic::PROPERTY_WRITE

```

**Comment by scheib** [comment]
Won't Fix, as noted by #241 the testing spec will not be updated until there is another implementation using it.
